packages/@webex/plugin-meetings/test/unit/spec/multistream/mediaRequestManager.ts

```diff
@@ -512,4 +519,155 @@ describe('MediaRequestManager', () => {
+    // set max macroblocks limit
+    mediaRequestManager.setDegradationPreferences({maxMacroblocksLimit: 32400});
+    assert.calledOnce(sendMediaRequestsCallback);
+  });
+
+  it('can degrade max-fs once when request exceeds max macroblocks limit', () => {
+    // set max macroblocks limit
+    mediaRequestManager.setDegradationPreferences({maxMacroblocksLimit: 32400});
+
+    // request 5 "large" 1080p streams
+    mediaRequestManager.addRequest(
+      {
+        policyInfo: {
+          policy: 'receiver-selected',
+          csi: 123,
+        },
+        receiveSlots: fakeReceiveSlots.slice(0, 5),
+        codecInfo: {
+          maxFs: 8192,
+        },
+      },
+      true
+    );
+
+    // check that resulting requests are 5 "medium" 720p streams
+    assert.calledWith(sendMediaRequestsCallback, [
+      sinon.match({
+        policy: 'receiver-selected',
+        policySpecificInfo: sinon.match({
+          csi: 123,
+        }),
+        receiveSlots: fakeWcmeSlots.slice(0, 5),
+        codecInfos: [
+          sinon.match({
+            payloadType: 0x80,
+            h264: sinon.match({
+              maxFs: 3600,
+            }),
+          }),
+        ],
+      }),
+    ]);
+  });
+
+  it('can degrade max-fs multiple times when request exceeds max macroblocks limit', () => {
+    // set max macroblocks limit
+    mediaRequestManager.setDegradationPreferences({maxMacroblocksLimit: 32400});
+
+    // request 10 "large" 1080p streams
+    mediaRequestManager.addRequest(
+      {
+        policyInfo: {
+          policy: 'receiver-selected',
+          csi: 123,
+        },
+        receiveSlots: fakeReceiveSlots.slice(0, 10),
+        codecInfo: {
+          maxFs: 8192,
+        },
+      },
+      true
+    );
+
+    // check that resulting requests are 10 "small" 360p streams
+    assert.calledWith(sendMediaRequestsCallback, [
+      sinon.match({
+        policy: 'receiver-selected',
+        policySpecificInfo: sinon.match({
+          csi: 123,
+        }),
+        receiveSlots: fakeWcmeSlots.slice(0, 10),
+        codecInfos: [
+          sinon.match({
+            payloadType: 0x80,
+            h264: sinon.match({
+              maxFs: 920,
+            }),
+          }),
+        ],
+      }),
+    ]);
+  });
+
+  it('can degrade only the largest max-fs when request exceeds max macroblocks limit', () => {
+    // set max macroblocks limit
+    mediaRequestManager.setDegradationPreferences({maxMacroblocksLimit: 32400});
+
+    // request 5 "large" 1080p streams and 5 "small" 360p streams
+    mediaRequestManager.addRequest(
+      {
+        policyInfo: {
+          policy: 'receiver-selected',
+          csi: 123,
+        },
+        receiveSlots: fakeReceiveSlots.slice(0, 5),
+        codecInfo: {
+          maxFs: 8192,
+        },
+      },
+      false
+    );
+    mediaRequestManager.addRequest(
+      {
+        policyInfo: {
+          policy: 'receiver-selected',
```

Comment:
minor thing, but all the tests are using "receiver-selected" policy. I know that policy is ignored in `getDegradedClientRequests()`, but maybe it would be worth putting "active-speaker" somewhere in one of the tests just to ensure that the code actually works no matter which policy is used.

Reply:
Sure, I'll change one of the tests to use "active-speaker".

Reply:
Sure, I'll change one of the tests to use "active-speaker".

Reply:
I ended up refactoring the unit tests to re-use some of the existing helper functions that were already there. The resulting file is a bit longer but I think it's much easier to read.